pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -189,7 +200,9 @@ func (h *handler) generateProvisioning(cluster *v3.Cluster, status v3.ClusterSta
 	}, status, nil
 }
 
-func (h *handler) generateCluster(cluster *v1.Cluster, status v1.ClusterStatus) ([]runtime.Object, v1.ClusterStatus, error) {
+// generateLegacyClusterFromProvisioningCluster will generate and return a clusters.management.cattle.io/v3 object based
+// on the passed in.
```

Comment:
Nit, but looks like it's missing reference to `clusters.provisioning.cattle.io/v1`
```suggestion
// on the clusters.provisioning.cattle.io/v1 object passed in.
```

Reply:
comment corrected!